doc/docs/development/system-tests.md

```diff
@@ -0,0 +1,195 @@
+
+
+## Robot test framework for system tests
+
+The [Robot test framework](https://robotframework.org/), often referred to as just "Robot Framework," is a popular open-source test automation framework used for automating test cases in various software applications. It is designed to be easy to use, highly readable, and adaptable for both beginners and experienced testers. It employs a keyword-driven approach, which means that test cases are written using a combination of keywords that represent actions, objects, and verifications. These keywords can be custom-defined by using Python programming language or come from libraries specific to the application under test. One of the standout features of Robot Framework is its human-readable syntax. Test cases are written in plain text composed with defined keywords, making it accessible to non-programmers and allowing stakeholders to understand and contribute to test case creation. Because of the ability to create custom keywords, a pool of domain specific and generic keywords could be defined to form a Ankaios project specific language for writing test cases.This makes it possible to directly use the test specifications written in natural language or the same wording of it to write automated test cases. This is the main reason why we used this test framework for system tests in Ankaios.
+
+## System tests structure
+```bash
+ankaios                              # Ankaios root
+  |--tests                           # Location for system tests and their resources
+  |  |--resources                    # Location for test resources
+  |  |  |--configs                   # Location for test case specific start-up configuration files
+  |  |  |  |--default.yaml           # A start-up configuration file
+  |  |  |  |--... <----------------  # Add more configuration files here!
+  |  |  |  
+  |  |  |--ankaios_library.py        # Ankaios keywords implementations
+  |  |  |--ankaios.resource          # Ankaios keywords
+  |  |  |--variables.resource        # Ankaios variables
+  |  |  |--... <-------------------  # Add more keywords and keywords implementation resources here!
+  |  |
+  |  |--stests                       # Location for system tests
+  |  |  |--workloads                 # Location for tests with specific test subject focus e.g. "workloads" for tests related "workloads"
+  |  |  |  |--list_workloads.robot   # A test suite testing "list workloads"
+  |  |  |  |--... <----------------  # Add more tests related to "workloads" here!
+  |  |  |... <---------------------  # Add test subject focus here!
+
+```
+
+## System test creation
+
+### A generic Ankaios system test structure
+The most common approach to create a robot test is using the space separated format where pieces of the data, such as keywords and their arguments, are separated from each others with two or more spaces. An alternative is using the pipe separated format where the separator is the pipe character surrounded with spaces (|).
```

Comment:
Shall we remove the alternative "|" syntax and only allow the "space" syntax? This would keep consistency.

Reply:
removed